### PR TITLE
Delete buttons for models

### PR DESCRIPTION
**Pull request by mcsgroi**
This is a follow up to https://github.com/Floppy/van_dam/pull/1073 which was closed in favor of opening two separate feature PRs.

Change summary:
- Add delete button to list and item view for collections

### CODE REVIEW

**Comment by mcsgroi** [comment]
@Floppy I've made the confirmation dialog adjustment to this delete PR as well that you mentioned before. Let me know if there's anything else you'd like changed.

**Comment by Floppy** [comment]
This does need a bit of extra logic to only delete in the right circumstances, but I need to think through that a bit. I'll tack it onto the end of this PR though and merge it in soon.

**Comment by Floppy** [comment]
So, I thought I'd done something safer when I added this for ModelFile, in https://github.com/Floppy/van_dam/commit/8039eaa8f8a1c26ea73373b627f59fc009f72aaf, but there's no safety; it just removes the file after destroy. AND, I misread this as thinking the file removal was in the model, not the controller. In summary, I'm a fool and this is fine to go in :)

File: app/controllers/models_controller.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require 'fileutils'
```
Comment (Floppy) [inline app/controllers/models_controller.rb:1]:
```suggestion
require "fileutils"
```

**Review by Floppy — APPROVED** [review]

**Pull request merged**